packaging/manager_test.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package packaging_test
+
+import (
+	"github.com/juju/juju/packaging"
+	coretesting "github.com/juju/juju/testing"
+	"github.com/juju/testing"
+	gc "gopkg.in/check.v1"
+)
+
+var _ = gc.Suite(&DependencyManagerTestSuite{})
+
+type DependencyManagerTestSuite struct {
+	coretesting.BaseSuite
+}
+
+func (s *DependencyManagerTestSuite) SetUpTest(c *gc.C) {
+	s.BaseSuite.SetUpTest(c)
+}
+
+func (s *DependencyManagerTestSuite) TestInstallWithCentos(c *gc.C) {
+	s.assertInstallCallsCorrectBinary(c, assertParams{
+		series:       "centos7",
+		pkg:          "foo",
+		pm:           packaging.YumPackageManager,
+		expPkgBinary: "yum",
+		expArgs: []string{
+			"--assumeyes", "--debuglevel=1", "install", "foo",
+		},
+	})
+}
+
+func (s *DependencyManagerTestSuite) TestInstallWithOpenSuse(c *gc.C) {
+	s.assertInstallCallsCorrectBinary(c, assertParams{
+		series:       "opensuseleap",
+		pkg:          "foo",
+		pm:           packaging.ZypperPackageManager,
+		expPkgBinary: "zypper",
+		expArgs: []string{
+			"--quiet", "--non-interactive", "install", "foo",
+		},
+	})
+}
+
+func (s *DependencyManagerTestSuite) TestInstallWithAptBackports(c *gc.C) {
+	s.assertInstallCallsCorrectBinary(c, assertParams{
+		series:       "trusty",
+		pkg:          "lxd", // requires backport on trusty
+		pm:           packaging.AptPackageManager,
+		expPkgBinary: "apt-get",
+		expArgs: []string{
+			"--option=Dpkg::Options::=--force-confold",
+			"--option=Dpkg::options::=--force-unsafe-io",
+			"--assume-yes", "--quiet", "install",
+			"--target-release", "trusty-backports", "lxd",
+		},
+	})
+}
+
+func (s *DependencyManagerTestSuite) TestInstallWithAptCloudTools(c *gc.C) {
+	s.assertInstallCallsCorrectBinary(c, assertParams{
+		series:       "precise",
+		pkg:          "mongodb-server", // requires backport on trusty
+		pm:           packaging.AptPackageManager,
+		expPkgBinary: "apt-get",
+		expArgs: []string{
+			"--option=Dpkg::Options::=--force-confold",
+			"--option=Dpkg::options::=--force-unsafe-io",
+			"--assume-yes", "--quiet", "install",
+			"--target-release", "precise-updates/cloud-tools",
+			"mongodb-server",
+		},
+	})
+}
+
+func (s *DependencyManagerTestSuite) TestInstallWithAptOnBionic(c *gc.C) {
+	s.assertInstallCallsCorrectBinary(c, assertParams{
+		series:       "bionic",
+		pkg:          "lxd",
+		pm:           packaging.AptPackageManager,
+		expPkgBinary: "apt-get",
+		expArgs: []string{
+			"--option=Dpkg::Options::=--force-confold",
+			"--option=Dpkg::options::=--force-unsafe-io",
```

Comment:
I know you mentioned this last week, but is `Dpkg::Options::` case insensitive?

Reply:
we should make it consistent, at least

Reply:
That comes from juju/packaging. I will submit a patch to nomalize this and update this PR to use the patched dependency.